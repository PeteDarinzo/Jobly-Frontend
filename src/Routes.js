import React, { useReducer } from "react";
import { Route, Redirect, Switch } from "react-router-dom";
import Home from "./Home";
import HomeAnon from "./HomeAnon";
import CompanyList from "./CompanyList";
import JobList from "./JobList";
import SignupForm from "./SignupForm";
import LoginForm from "./LoginForm";
import CompanyDetail from "./CompanyDetail";
import ProfileForm from "./ProfileForm";

const Routes = ({ companies, jobs, register, login, updateUser, apply, filter, clearFilter, applications, loggedIn, userCredentials, loginError, signupError, profileError }) => {

  return (
    <Switch>
      <Route exact path="/">
        {loggedIn
          ? <Home username={userCredentials.username} />
          : <HomeAnon />}
      </Route>
      <Route exact path="/companies">
        {loggedIn
          ? <CompanyList companies={companies} filter={filter} clearFilter={clearFilter} />
          : <Redirect to="/login" />}
      </Route>
      <Route exact path="/companies/:handle">
        {loggedIn
          ? <CompanyDetail companies={companies} jobs={jobs} apply={apply} applications={applications} />
          : <Redirect to="/login" />}
      </Route>
      <Route exact path="/jobs">
        {loggedIn
          ? <JobList jobs={jobs} apply={apply} applications={applications} />
          : <Redirect to="/login" />}
      </Route>
      <Route exact path="/profile">
        <ProfileForm
          updateUser={updateUser}
          username={userCredentials.username}
          firstName={userCredentials.firstName}
          lastName={userCredentials.lastName}
          email={userCredentials.email}
          error={profileError} />
      </Route>
      <Route exact path="/signup" >
        <SignupForm register={register} error={signupError} />
      </Route>
      <Route exact path="/login">
        <LoginForm login={login} error={loginError} />
      </Route>
      <Redirect to="/" />
    </Switch>
  );

}

export default Routes;