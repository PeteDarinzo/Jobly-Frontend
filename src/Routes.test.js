import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Routes from "./Routes";


it("renders routes without crashing", () => {
  render(
    <MemoryRouter>
      <Routes companies={[]} jobs={[]} register={() => { }} login={() => { }} updateUser={() => { }} apply={() => { }} filter={() => { }} clearFilter={() => { }} applications={[]} loggedIn="" userCredentials={{}} loginError="" signupError="" profileError="" />
    </MemoryRouter>
  )
});

it("renders matches the router snapshot when logged out", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Routes companies={[]} jobs={[]} register={() => { }} login={() => { }} updateUser={() => { }} apply={() => { }} filter={() => { }} clearFilter={() => { }} applications={[]} loggedIn="" userCredentials={{}} loginError="" signupError="" profileError="" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders matches the router snapshot when logged in", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Routes companies={[]} jobs={[]} register={() => { }} login={() => { }} updateUser={() => { }} apply={() => { }} filter={() => { }} clearFilter={() => { }} applications={[]} loggedIn="test-token" userCredentials={{}} loginError="" signupError="" profileError="" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
