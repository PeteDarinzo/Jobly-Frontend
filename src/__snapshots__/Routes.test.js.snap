// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders matches the router snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="Home shadow-lg"
  >
    <h1>
      Welcome back, !
    </h1>
    <a
      href="/jobs"
    >
      <button
        class="m-3 btn btn-primary btn-lg"
      >
        View Jobs
      </button>
    </a>
    <a
      href="/companies"
    >
      <button
        class="m-3 btn btn-primary btn-lg"
      >
        View Companies
      </button>
    </a>
  </div>
</DocumentFragment>
`;

exports[`renders matches the router snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Home shadow-lg"
  >
    <h1>
      Welcome to Jobly!
    </h1>
    <h2
      class="Home-left"
    >
      Your dream job...
    </h2>
    <a
      href="/signup"
    >
      <button
        class="m-3 btn btn-primary btn-lg"
      >
        Sign Up
      </button>
    </a>
    <a
      href="/login"
    >
      <button
        class="m-3 btn btn-primary btn-lg"
      >
        Log In
      </button>
    </a>
    <h2
      class="Home-right"
    >
      ...just a click away.
    </h2>
  </div>
</DocumentFragment>
`;
